Source/WebCore/accessibility/AccessibilityNodeObject.cpp

```diff
@@ -269,7 +269,7 @@ LayoutRect AccessibilityNodeObject::checkboxOrRadioRect() const
 
 LayoutRect AccessibilityNodeObject::elementRect() const
 {
-    if (isCheckboxOrRadio())
+    if (isCheckboxOrRadio() || isSwitch())
         return checkboxOrRadioRect();
```

Comment:
The purpose `checkboxOrRadioRect()` is to unify the rect of the control with the rect of its associated label. But right now this only works with native `<label for="id-of-control">Foo label</label>`, or containing labels:

```html
<label>
   Foo bar label for switch
   <div role="switch"></div>
</label>
```

But I'm wondering if either of these methods of pairing labels are "allowed" or recommended patterns for a `role="switch"`? Certainly for `<input type=checkbox switch>`, but `role="switch"` I'm less confident in.

@cookiecrook what do you think?

Reply:
Hmm, but the logic here also does this for `<label><span role=checkbox></label>`, right? Perhaps it shouldn't be using `isCheckboxOrRadio()` to begin with?

Reply:
Yes you're right, it would behave that way for role=checkbox, but arguably shouldn't. Let's remove this part and we'll address it separately later.

Reply:
#19048

Reply:
Yes label for/id should only work on native inputs, not AIRA controls.